File: Data Structures & Algorithms/trapping-rain-water/submission-1.rs
impl Solution {
pub fn trap(height: Vec<i32>) -> i32 {
    let mut res = 0;

    let mut l = 0;
    let mut lh = height[l];
    let mut r = height.len() - 1;
    let mut rh = height[r];
    while l < r {
        if lh < rh {
            l += 1;
            lh = lh.max(height[l]);
            res += lh - height[l];
        } else {
            r -= 1;
            rh = rh.max(height[r]);
            res += rh - height[r];
        }
    }
    res
}

}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.cpp
class Solution {
 public:
  int maxDepth(TreeNode* root) {
    if(!root) return 0;
    int res = 0;
    deque<pair<TreeNode*, int>> v{{root, 1}};
    while (!v.empty()) {
      auto& [curr, level] = v.front();
      res = max(res, level);
      if (curr->left) v.push_back({curr->left, level + 1});
      if (curr->right) v.push_back({curr->right, level + 1});
      v.pop_front();
    }
    return res;
  }
};
